katdal/datasources.py

```diff
@@ -247,6 +248,37 @@ def _shorten_key(telstate, key):
     return ''
 
 
+def _infer_chunk_store(url_parts, telstate, chunk_name=None,
+                       s3_endpoint_url=None, **kwargs):
+    """Construct chunk store automatically from dataset URL and telstate."""
+    data_path = store_path = ''
+    try:
+        if url_parts.scheme == 'file':
+            # Look for adjacent data directory (presumably containing NPY files)
+            rdb_path = os.path.abspath(url_parts.path)
+            store_path = os.path.dirname(os.path.dirname(rdb_path))
+            if not store_path:
+                store_path = os.path.curdir
+            try:
+                chunk_name = chunk_name or telstate['chunk_name']
+            except KeyError as e:
+                raise StoreUnavailable('[{}] {}'.format(type(e).__name__, e))
+            else:
+                data_path = os.path.join(store_path, chunk_name)
+        if os.path.isdir(data_path):
+            return NpyFileChunkStore(store_path)
+        try:
+            s3_endpoint_url = s3_endpoint_url or telstate['s3_endpoint_url']
+        except KeyError as e:
+            raise StoreUnavailable('[{}] {}'.format(type(e).__name__, e))
+        else:
+            return S3ChunkStore.from_url(s3_endpoint_url)
+    except StoreUnavailable as e:
+        logger.warn('StoreUnavailable: %s', e)
+        logger.warn('Chunk store unavailable - opening metadata only')
```

Comment:
Maybe just log one line e.g. `Chunk store unavailable (%s) - opening metadata only`.

Reply:
I had it exactly like that but did not like the output. The %s can be quite long and an eye-strain, making one miss the important bit at the end (`opening metadata only`).

Reply:
Maybe move the %s to the end then? My thinking is that Elasticsearch treats each log message as a document, so when split into two lines like this you get two documents which could end up interleaved with log messages from other processes etc.